src/fgetattrlist.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2024 Mihai Moldovan <ionic@ionic.de>, raf <raf@raf.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/* MP support header */
+#include "MacportsLegacySupport.h"
+#if __MP_LEGACY_SUPPORT_FSETATTRLIST__
+
+#include <sys/attr.h>
+#include <sys/fcntl.h>
+#include <sys/param.h>
+#include <sys/errno.h>
+
+#include <fcntl.h>
+#include <string.h>
+#include <unistd.h>
+
+
+#ifdef __LP64__
+int fgetattrlist(int fd, void *a, void *buf, size_t size, unsigned int flags)
+#else /* defined (__LP64__) */
+int fgetattrlist(int fd, void *a, void *buf, size_t size, unsigned long flags)
+#endif /* defined (__LP64__) */
+{
+    int cont = 1,
+        ret = 0;
+
+    char fpath[MAXPATHLEN];
+    memset (fpath, 0, MAXPATHLEN);
+    if (-1 == fcntl(fd, F_GETPATH, fpath)) {
+        ret = EBADF;
+        cont = 0;
+    }
+
+#if __MPLS_TARGET_OSVER < 1080
+    /*
+     * Older systems don't correctly check if no attributes are to be set, which usually
```

Comment:
Copy-'n-paste error: "set" isn't really correct here.  Maybe "read" or "obtained".

Based on the description, I don't know if the problem addressed here exists on reads at all, though as you say, the "fix" probably doesn't hurt.

Reply:
Done